### ISSUE TITLE
ClusterServer: channel closed error

### ISSUE THREAD

**Issue by endel** [gamestdio/colyseus#126]
Not sure what's the impact of this error. It occurs randomly on my game.

Looks like an Inter-process communication issue from [memshared](https://github.com/endel/memshared).

**Error:**

```
events.js:160
      throw er; // Unhandled 'error' event
      ^
Error: channel closed
    at process.target.send (internal/child_process.js:540:16)
    at Store.dispatch (/root/crashracing-production/build/node_modules/memshared/lib/Store.js:20:17)
    at Object.sismember (/root/crashracing-production/build/node_modules/memshared/lib/commands/set.js:74:18)
    at WebSocket.<anonymous> (/root/crashracing-production/build/node_modules/colyseus/lib/matchmaking/Process.js:54:19)
    at emitOne (events.js:96:13)
    at WebSocket.emit (events.js:188:7)
    at Receiver._receiver.onmessage (/root/crashracing-production/build/node_modules/ws/lib/WebSocket.js:141:47)
    at Receiver.dataMessage (/root/crashracing-production/build/node_modules/ws/lib/Receiver.js:380:14)
    at Receiver.getData (/root/crashracing-production/build/node_modules/ws/lib/Receiver.js:330:12)
    at Receiver.startLoop (/root/crashracing-production/build/node_modules/ws/lib/Receiver.js:165:16)
```

**Happened on this call:**

https://github.com/gamestdio/colyseus/blob/3c16ee7604d46801727a132d3678d998ba590ac1/src/matchmaking/Process.ts#L67

**Comment by endel** [comment]
Apparently this may be happening due to match-making process not being shut down properly: https://stackoverflow.com/a/24128164

**Comment by BitOfGold** [comment]
this (or a similar looking error) happens to me with Server too. No ClusterServer

`gameServer = new colyseus.Server(`

I have two browser windows automatically refreshed at the sam time by livereloadx on changes (client)
I use nodemon to refresh on changes (server)

Server crashes (repeated) with the following message.
If I start the server (node --harmony ./index.js), then refrest first client and then the other client, it works OK.
Very frustrating :(

`events.js:183
      throw er; // Unhandled 'error' event
      ^

Error: read ECONNRESET
    at _errnoException (util.js:1022:11)
    at TCP.onread (net.js:615:25)
[nodemon] app crashed - waiting for file changes before starting...
[nodemon] restarting due to changes...
[nodemon] starting `node --harmony ./index.js localhost 3553`
'uws' not installed. using 'ws' instead.
Listening on http://localhost:4236
GameRoom created! { requestId: 1,
  clientId: 'Byu7B-wOf',
  sessionId: 'rkfWqbOOz',
  op1: 'test' }
events.js:183
      throw er; // Unhandled 'error' event
      ^

Error: read ECONNRESET
    at _errnoException (util.js:1022:11)
    at TCP.onread (net.js:615:25)
[nodemon] app crashed - waiting for file changes before starting...
`

**Comment by endel** [comment]
HI @BitOfGold, the error you're having is not the same as the issue here. Which versions of node and `colyseus` are you using? 

I'd recommend using the latest version of colyseus (`0.8.13`) and a node version `6.0.0` or higher.

Cheers

**Comment by BitOfGold** [comment]
Hey @endel,
Thanks, problem solved with new version 0.8.13 and client version 0.8.2

(Thank you for this great library!)

**Comment by endel** [comment]
`ClusterServer` has been deprecated and this issue shouldn't happen anymore (https://github.com/gamestdio/colyseus/pull/135)
